### PR TITLE
Scheduled Reports: Enable security groups support and add the subpanel

### PR DESCRIPTION
**Pull request by lazka**
## Description

Without security groups when a user creates a new scheduled report then no
other user in the same security group can see/edit the record.

This adds security group support to the module and also adds a security groups
subpanel in the detail view so the group relations can be edited.

## Motivation and Context

I'd like to use scheduled reports in combination with security groups.

## How To Test This

* (Run a repair to adjust the relations)
* Create a scheduled report with a user that is in a security group
* Log in as a different user in the same group and see/edit the scheduled report

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist
- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [ ] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

File: modules/AOR_Scheduled_Reports/metadata/subpaneldefs.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+if (!defined('sugarEntry') || !sugarEntry) {
+    die('Not A Valid Entry Point');
+}
+/**
+ *
+ * SugarCRM Community Edition is a customer relationship management program developed by
+ * SugarCRM, Inc. Copyright (C) 2004-2013 SugarCRM Inc.
+ *
+ * SuiteCRM is an extension to SugarCRM Community Edition developed by SalesAgility Ltd.
+ * Copyright (C) 2011 - 2018 SalesAgility Ltd.
```
Comment (Dillon-Brown) [inline modules/AOR_Scheduled_Reports/metadata/subpaneldefs.php]:
```suggestion
 * Copyright (C) 2011 - 2019 SalesAgility Ltd.
```

File: modules/AOR_Scheduled_Reports/metadata/subpaneldefs.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+
+$module_name = 'AOR_Scheduled_Reports';
+$layout_defs[$module_name] = array(
```
Comment (Dillon-Brown) [inline modules/AOR_Scheduled_Reports/metadata/subpaneldefs.php]:
```suggestion
$layout_defs[$module_name] = [
```

File: modules/AOR_Scheduled_Reports/metadata/subpaneldefs.php [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    'subpanel_setup' => array(
```
Comment (Dillon-Brown) [inline modules/AOR_Scheduled_Reports/metadata/subpaneldefs.php]:
```suggestion
    'subpanel_setup' => [
```

File: modules/AOR_Scheduled_Reports/metadata/subpaneldefs.php [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        'securitygroups' => array(
```
Comment (Dillon-Brown) [inline modules/AOR_Scheduled_Reports/metadata/subpaneldefs.php]:
```suggestion
        'securitygroups' => [
```

File: modules/AOR_Scheduled_Reports/metadata/subpaneldefs.php [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            'top_buttons' => array(array('widget_class' => 'SubPanelTopSelectButton', 'popup_module' => 'SecurityGroups', 'mode' => 'MultiSelect'),),
```
Comment (Dillon-Brown) [inline modules/AOR_Scheduled_Reports/metadata/subpaneldefs.php]:
```suggestion
            'top_buttons' => [['widget_class' => 'SubPanelTopSelectButton', 'popup_module' => 'SecurityGroups', 'mode' => 'MultiSelect'],],
```

File: modules/AOR_Scheduled_Reports/metadata/subpaneldefs.php [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+            'order' => 900,
+            'sort_by' => 'name',
+            'sort_order' => 'asc',
+            'module' => 'SecurityGroups',
+            'refresh_page' => 1,
+            'subpanel_name' => 'default',
+            'get_subpanel_data' => 'SecurityGroups',
+            'add_subpanel_data' => 'securitygroup_id',
+            'title_key' => 'LBL_SECURITYGROUPS_SUBPANEL_TITLE',
+        ),
```
Comment (Dillon-Brown) [inline modules/AOR_Scheduled_Reports/metadata/subpaneldefs.php]:
```suggestion
        ],
```

File: modules/AOR_Scheduled_Reports/metadata/subpaneldefs.php [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    ),
```
Comment (Dillon-Brown) [inline modules/AOR_Scheduled_Reports/metadata/subpaneldefs.php]:
```suggestion
    ],
```

File: modules/AOR_Scheduled_Reports/metadata/subpaneldefs.php [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+);
```
Comment (Dillon-Brown) [inline modules/AOR_Scheduled_Reports/metadata/subpaneldefs.php]:
```suggestion
];
```

File: modules/AOR_Scheduled_Reports/vardefs.php [hunk 9]
```diff
@@ -143,4 +143,4 @@
 if (!class_exists('VardefManager')) {
     require_once('include/SugarObjects/VardefManager.php');
 }
-VardefManager::createVardef('AOR_Scheduled_Reports', 'AOR_Scheduled_Reports', array('basic'));
+VardefManager::createVardef('AOR_Scheduled_Reports', 'AOR_Scheduled_Reports', array('basic', 'security_groups'));
```
Comment (Dillon-Brown) [inline modules/AOR_Scheduled_Reports/vardefs.php]:
```suggestion
VardefManager::createVardef('AOR_Scheduled_Reports', 'AOR_Scheduled_Reports', ['basic', 'security_groups']);
```

**Comment by Mac-Rae** [comment]
Assessed :+1:

**Pull request merged**